kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/ProxyRpcTest.java

```diff
@@ -90,6 +91,11 @@ void testKroxyliciousCanDecodeManipulateAndProxyRPC(Scenario scenario) {
     @NonNull
     private static Stream<Scenario> scenarios() {
         Map<ApiAndVersion, ApiMessage> requestSamples = ApiMessageSampleGenerator.createRequestSamples();
+        // Ensure that all produce requests require a response.
+        requestSamples.values().stream()
```

Comment:
This issue was causing a spurious test fail when the generated test data resulted in a produce request with acks=0.  It is not strictly speaking anything to do with 3.9, but for some reason with the extra RPCs included in 3.9, it failed consistently for me.

Reply:
ApiMessageSampleGenerator is a bit of a brute force thing that populates all message fields with random data, the message semantics are a bit out of it's wheelhouse so far. 

The generation is randomised but seeded to try and have a deterministic result. When generating the samples it's ordering ApiKeys by name, so I guess ProduceRequest has changed position and is now unluckily getting a `0` generated for acks.

I think we should move Keith's logic into `createRequestSamples` and doc the method as returning only requests that have responses, since that's what it's been targeting (or rename it, naming is hard though, responsefulRequestSamples? RequestWithResponseSamples?).

Reply:
Ohh I hadn't realised it was there I wonder if/how we should combine it with [RequestFactory](https://github.com/kroxylicious/kroxylicious/blob/d86ed24a6a2efd33d5881cecbeae1c29a5593592/kroxylicious-filter-test-support/src/main/java/io/kroxylicious/test/RequestFactory.java).

Thats clearly out of scope for this PR.